app/models/book.rb

```diff
@@ -1,2 +1,7 @@
 class Book < ApplicationRecord
+  belongs_to :category
+  has_many :requests, dependent: :nullify
```

Comment:
ủa sao chỗ này lại dùng `nullify` em nhỉ

Reply:
chỗ này e dùng `nullify` ý tưởng là khi xóa 1 book thì vẫn giữ lại các requests đã có...gán foreign key thành null
mà e mới check lại thì ko đc vì có ràng buộc not null
vậy cho e hỏi chỗ này nên sử lý như thế nào anh

Reply:
em tìm hiểu về option` optional: true` trong rails nhé

Reply:
tiện nếu như em muốn xóa 1 book mà muốn giữ nguyên các request thì có thể sử dụng soft- delete